agent/acs/handler/payload_handler_test.go

```diff
@@ -370,16 +370,16 @@ func TestAddPayloadTaskAddsNonStoppedTasksAfterStoppedTasks(t *testing.T) {
 	if firstTaskAdded.Arn != stoppedTaskArn {
 		t.Errorf("Expected first task arn: %s, got: %s", stoppedTaskArn, firstTaskAdded.Arn)
 	}
-	if firstTaskAdded.DesiredStatus != api.TaskStopped {
-		t.Errorf("Expected first task state be be: %s , got: %s", "STOPPED", firstTaskAdded.DesiredStatus.String())
+	if firstTaskAdded.DesiredStatusUnsafe != api.TaskStopped {
```

Comment:
This really points out how our test code is doing things different from production code.  This should probably call `GetDesiredStatus` instead.